src/main/java/max/storage/Storage.java

```diff
@@ -0,0 +1,115 @@
+package max.storage;
+
+import max.commands.*;
+import max.exception.MaxException;
+import max.parser.Parser;
+import max.storage.Storage;
+import max.tasks.TaskList;
+import max.tasks.*;
+import max.ui.Ui;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * Loads and saves task list to memory.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Initialises location of task list in memory.
+     *
+     * @param filePath
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads task list from memory.
+     *
+     * @return List of tasks
+     * @throws MaxException If file input cannot be interpreted
+     */
+    public ArrayList<Task> load() throws MaxException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                loadTask(line, tasks);
+            }
+        } catch (IOException e) {
+//            e.printStackTrace();
+        }
+        return tasks;
+    }
+
+    /**
+     * Reads file input and adds task to task list.
+     *
+     * @param task String representation of task stored in memory
+     * @param tasks Task list
+     * @throws MaxException If String representation of task is invalid
+     */
+    public void loadTask(String task, ArrayList<Task> tasks) throws MaxException {
+        String type = task.substring(0, 1).trim(); // T, D, E
+        boolean isDone = task.substring(3, 5).trim().equals("1");
+
+        switch (type) {
+        case "T":
+            String description = task.substring(7).trim();
+            tasks.add(new Todo(description, isDone));
+            break;
+        case "D": {
+            int byIndex = task.indexOf("by:");
+            String item = task.substring(7, byIndex - 1).trim();
+            String by = task.substring(byIndex + 3).trim();
+            LocalDate byDate = LocalDate.parse(by);
+            tasks.add(new Deadline(item, byDate, isDone));
+            break;
+        }
+        case "E": {
+            int fromIndex = task.indexOf("from:");
+            int toIndex = task.indexOf("to:");
+            String item = task.substring(7, fromIndex - 1).trim();
+            String from = task.substring(fromIndex + 5, toIndex - 1).trim();
+            String to = task.substring(toIndex + 3).trim();
+            LocalDate fromDate = LocalDate.parse(from);
+            LocalDate toDate = LocalDate.parse(to);
+            tasks.add(new Event(item, fromDate, toDate, isDone));
+            break;
+        }
+        default:
+            throw new MaxException("Hey! Your storage file is corrupted.");
+        }
+    }
```

Comment:
Consider replacing the "magic numbers" such as 1, 3, 5 and 7.  Maybe `String.split()` would be more useful here than `substring()`.

Reply:
Thanks for the feedback! I will update this in future commits :)